p2p/enode/node.go

```diff
@@ -200,6 +200,16 @@ func (n *Node) TCPEndpoint() (netip.AddrPort, bool) {
 	return netip.AddrPortFrom(n.ip, n.tcp), true
 }
 
+// QUICEndpoint returns the announced QUIC endpoint.
+func (n *Node) QUICEndpoint() (netip.AddrPort, bool) {
+	var quic enr.QUIC
+	n.Load(&quic)
+	if !n.ip.IsValid() || n.ip.IsUnspecified() || quic == 0 {
```

Comment:
Here we need to distinguish if the IP is IPv6, and try loading QUIC6 if it's there.

Reply:
Just addressed it. Let me know if it looks right